### PR TITLE
Implementation of app helper macros

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at providing helpful macros to avoid having to rewrite a lot of code when creating an abstract app.

You can test the implementations here : https://github.com/AbstractSDK/app-template/pull/6

This PR also updates cw_storage_plus and trait implementation on references (fixed for CI)

## This Includes
- [x] an interface macro for defining the cw-orch interface automatically
- [ ] AppExecuteMsgFns
- [x] app_messages!

### CODE REVIEW

File: packages/abstract-app/src/interface.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+	    pub mod interface{
+	    	use super::_wrapper_fns;
+	    	#[::cw_orch::interface(_wrapper_fns::InstantiateMsg, _wrapper_fns::ExecuteMsg, _wrapper_fns::QueryMsg, _wrapper_fns::MigrateMsg)]
+			pub struct $app_type;
+
+
+			impl <Chain: ::cw_orch::prelude::CwEnv> ::cw_orch::prelude::Uploadable for $app_type<Chain> {
+			    fn wasm(&self) -> ::cw_orch::prelude::WasmPath {
+			        ::cw_orch::prelude::ArtifactsDir::auto(Some(env!("CARGO_MANIFEST_DIR").to_string())).find_wasm_path(env!("CARGO_CRATE_NAME")).unwrap()
```
Comment (CyberHoward) [inline packages/abstract-app/src/interface.rs:77]: I believe that the optimizer changes '-' to '_' so this name won't be found.

File: packages/abstract-app/src/msgs.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#[macro_export]
+/// Groups code that is needed on every app.
+/// This registers the types for safety when using Messages
+/// This is also used to indicate that The Query And Execute messages or used as app messages
+macro_rules! app_messages {
```
Comment (CyberHoward) [inline packages/abstract-app/src/msgs.rs:5]: Can you apply this macro in the example in `abstract-app/examples/counter.rs`?

**Review by CyberHoward — CHANGES_REQUESTED** [review]

File: packages/abstract-app/src/interface.rs [hunk 3]
```diff
@@ -0,0 +1,100 @@
+#[macro_export]
+/// Creates the interface for working with the app wth cw-orch
+macro_rules! create_interface {
```
Comment (CyberHoward) [inline packages/abstract-app/src/interface.rs:3]: can you apply this in the example as well?

File: packages/abstract-app/src/msgs.rs (hunk 2 above)
Comment (Kayanski) [reply packages/abstract-app/src/msgs.rs:5]: https://github.com/AbstractSDK/contracts/pull/371/commits/d919315ae6b9ce0eaccaf378f552d33421ff9b83

File: packages/abstract-app/src/interface.rs (hunk 3 above)
Comment (Kayanski) [reply packages/abstract-app/src/interface.rs:3]: https://github.com/AbstractSDK/contracts/pull/371/commits/60e6ec116288334c5c343c5721a89408cf973d77

File: packages/abstract-app/src/interface.rs (hunk 1 above)
Comment (Kayanski) [reply packages/abstract-app/src/interface.rs]: https://github.com/AbstractSDK/contracts/pull/371/commits/88729b1028a5d3d3579713257362f46aa0f0065c

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**